torch/testing/_internal/common_utils.py

```diff
@@ -1469,10 +1469,10 @@ def assertWarnsOnceRegex(self, category, regex=''):
                 torch.set_warn_always(prev)
             if len(ws) == 0:
                 self.fail('no warning caught')
-            for w in ws:
-                self.assertTrue(type(w.message) is category)
-                self.assertTrue(re.match(pattern, str(w.message)),
-                                f'{pattern}, {w.message}')
+            self.assertTrue(any([type(w.message) is category for w in ws]))
+            self.assertTrue(
+                any([re.match(pattern, str(w.message)) for w in ws]),
```

Comment:
I really like the motivation for this change. 

The second argument to assertTrue is a string message that appears to a developer when the assertion fails, right? In this case it'd be something like "foo string, warning message warning message warning message" if there were multiple warning messages? 

While warning messages don't need to be perfect, what do you think about filtering ws using the `type(w.message) is category` check above? That way the warning messages will be from the set that should be matched against, and the previous assertion could just validate this filtered set is nonempty.